pkg/ingester/client/client.go

```diff
@@ -42,3 +43,15 @@ func MakeIngesterClient(addr string, withCompression bool) (IngesterClient, erro
 func (c *closableIngesterClient) Close() error {
 	return c.conn.Close()
 }
+
+type Config struct {
+	MaxRecvMsgSize int
+	CompressToIngester bool
+}
+
+func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
+	// We have seen 20MB returns from queries - add a bit of headroom
+	f.IntVar(&cfg.MaxRecvMsgSize, "ingester.client.max-recv-message-size", 64*1024*1024, "Maximum message size, in bytes, this client will receive.")
+	// moved from distributor pkg, but flag prefix left as-is so existing users do not break.
+	flag.BoolVar(&cfg.CompressToIngester, "distributor.compress-to-ingester", false, "Compress data in calls to ingesters.")
```

Comment:
Rather than the confusing name, I think I'd rather have two flags and copy from one to the other to allow backwards-compatibility.

Reply:
Certainly. I've added this in the latest updates to the PR.